pl_bolts/models/self_supervised/simclr/simclr_module.py

```diff
@@ -74,197 +87,371 @@ def __init__(self,
         super().__init__()
         self.save_hyperparameters()
 
-        self.nt_xent_loss = nt_xent_loss
-        self.encoder = self.init_encoder()
-
-        # h -> || -> z
-        self.projection = Projection()
-
-    def init_encoder(self):
-        encoder = resnet50_bn(return_all_feature_maps=False)
-
-        # when using cifar10, replace the first conv so image doesn't shrink away
-        encoder.conv1 = nn.Conv2d(
-            3, 64,
-            kernel_size=3,
-            stride=1,
-            padding=1,
-            bias=False
+        self.gpus = gpus
+        self.nodes = nodes
+        self.arch = arch
+        self.dataset = dataset
+        self.num_samples = num_samples
+        self.batch_size = batch_size
+
+        self.hidden_mlp = hidden_mlp
+        self.feat_dim = feat_dim
+        self.first_conv = first_conv
+        self.maxpool1 = maxpool1
+
+        self.optim = optimizer
+        self.lars_wrapper = lars_wrapper
+        self.exclude_bn_bias = exclude_bn_bias
+        self.weight_decay = weight_decay
+        self.temperature = temperature
+
+        self.start_lr = start_lr
+        self.final_lr = final_lr
+        self.learning_rate = learning_rate
+        self.warmup_epochs = warmup_epochs
+        self.max_epochs = max_epochs
```

Comment:
why do you need this when you already have it in `self.hparams.my_arg`?

Reply:
Again, don't like the idea of using self.hparams.something

Reply:
well, then you have the parameters in two places `self.my_arg` and `self.hparams.my_arg` which is confusing for any user... I think that in such a case you should start the discussion in PL instead of changing it here in Bolts :]

Reply:
Hey both ! I understand Jirka point, but I also find using hparams everything is little bit painful.

Can't we just do something like:

```
for attr_name, attr_value in self.hparams.items():  
    existing_attr = getattr(self, attr_name, None)
    if existing_attr:
        setattr(self, attr_name, attr_value)
    else:
        warning .... use self.hparams.attr_name as attr_name already exists in the model
```

Reply:
well, so what is the motivation for doing so?
1. have attributes directly in `self`
2. do assignment/init in `__init__` so it is well indexed by IDE
3. have freedom do change the attribute name
4. variable recapitulation

Reply:
Let's bring this up on the PL side, it's fine here for now (as a design choice) but we need to discuss this further because I don't like hparams either.